vm-virtio/src/net_util.rs

```diff
@@ -47,6 +73,190 @@ pub enum Error {
     TapEnable(TapError),
 }
 
+pub struct CtrlVirtio {
+    pub queue_evt: EventFd,
+    pub queue: Queue,
+}
+
+impl std::clone::Clone for CtrlVirtio {
+    fn clone(&self) -> Self {
+        CtrlVirtio {
+            queue_evt: self.queue_evt.try_clone().unwrap(),
+            queue: self.queue.clone(),
+        }
+    }
+}
+
+impl CtrlVirtio {
+    pub fn new(queue: Queue, queue_evt: EventFd) -> Self {
+        CtrlVirtio { queue_evt, queue }
+    }
+
+    fn process_mq(&self, mem: &GuestMemoryMmap, avail_desc: DescriptorChain) -> Result<()> {
```

Comment:
Adding the MQ handling to the control queue should be a in a different commit as it doesn't do anything without the rest of the MQ implementation.

Reply:
Ah, yes, right, I will split it into another commit.

Reply:
@rbradford I think about this again, actually, here, control queue is added to work for MQ, it does not mean so much to separate MQ implementation from control queue, however, I will update the commit message to indicate that control queue is the pre-condition for mq support for net devices.

Reply:
I move the mq process into a separate commit finally, since I think there are more details I need to explain for that.